### ISSUE TITLE
Parallax2D: Scroll Scale is not respecting the Y value

### ISSUE THREAD

**Issue by ElvisVilla** [godotengine/godot#93054]
### Tested versions

- Reproducible in Godot 4.3 Beta 1

### System information

MacOs Sonoma 14.5 - Godot 4.3 Beta 1

### Issue description

In my project I have some Parallax2D and im modifying the Scroll Scale for custom speed on each Parallax2D. Its working well on the X component, the parallax moment on horizontal works well, but the vertical movement is not properly being compute, the trees on the video below should not follow the camera if the Scroll Scale is 0 value on Y, but it does it, and if the value is 0.1 it should follow slowly. This is how i understand the value multiplier should behave. 

I would expect the Y component affect the speed on vertical for different values as it says on the documentation:

> Multiplier to the final Parallax2D's offset. Can be used to simulate distance from the camera.
> 
> For example, a value of 1 scrolls at the same speed as the camera. A value greater than 1 scrolls faster, making objects appear closer. Less than 1 scrolls slower, making objects appear further, and a value of 0 stops the objects completely.

The Scroll Scale on X work as intented but the Y component is not calculating the speed on vertical. Even when using different values the speed of Y on the Scroll Scale remain the same, it doesn't look to change at all. From simple observation my guess could be one of this 2 reasons:

1.  Looks like the Values on Y component are not being computed properly or being ignored.
2. The Lock / Unlock component ratio that allows to set different components on Vector2 properties on the Inspector is messing up with the final values.

I dont have any specific code attached to those Nodes, im only using the default behaviour of the new Parallax2D. 

<img width="1552" alt="Captura de pantalla 2024-06-11 a la(s) 1 04 20 p  m" src="https://github.com/godotengine/godot/assets/37193450/5d179df4-a156-4eaa-a8c1-844398960aa1">

https://github.com/godotengine/godot/assets/37193450/206c804d-ae55-4a9b-95cc-5ba6b9ae49ee

### Steps to reproduce

N/A

### Minimal reproduction project (MRP)

N/A

**Comment by markdibarry** [comment]
I'm unable to replicate this. It seems to be working fine for me.


https://github.com/godotengine/godot/assets/21325943/c6bb73f2-b8c7-412a-bcc1-cb460235a5ae


Can you make an MRP for me to test?

**Comment by roydbt** [comment]
[MRP.zip](https://github.com/user-attachments/files/15849257/MRP.zip)
Here is an MRP for the same issue. Moving along the x-axis scrolls fine, but no scrolling at all across the y-axis unless I specify a repeat size for it.

**Comment by kleonc** [comment]
> [MRP.zip](https://github.com/user-attachments/files/15849257/MRP.zip)
Here is an MRP for the same issue. Moving along the x-axis scrolls fine, but no scrolling at all across the y-axis unless I specify a repeat size for it.

From the MRP's `main.tscn`:
```
[node name="ParallaxLayer" type="Parallax2D" parent="."]
scroll_scale = Vector2(0.5, 1)
repeat_size = Vector2(1024, 2.08165e-12)
```

So this is likely the same as #91631.

Can't confirm the same about the OP (no MRP provided), but seems all reporters run on Mac. :thinking: I can't reproduce this myself / debug in detail / pinpoint the cause.

**Comment by markdibarry** [comment]
@kleonc Just tested the MRP, and you're right. It's not related to Parallax2D, and is the issue on mac with editor inputs mentioned in the other ticket. Maybe we should update that issue's title as well.

**Comment by roydbt** [comment]
Currently, the Godot editor is unusable on macOS (I have to close the editor, replace every 2.08165e-12 with 0 manually, and then restart the engine to make these changes stick). How can I help fix this issue?

**Comment by markdibarry** [comment]
Should also be closed by #94655 since it was the same root cause.
